shtdlib.sh

```diff
@@ -325,10 +328,44 @@ function version_sort {
         local vsorter='sort -t. -k1,1n -k2,2n -k3,3n -k4,4n'
     fi
 
-    for arg in "${@}${piped_versions[@]}" ; do
+    for arg in "${@}" ; do
         echo "${arg}"
     done | ${vsorter}
 }
+# shellcheck disable=2120
+function version_sort {
+    # First command needs to be read, this way any piped input goes to it
+    while read -rt "${read_timeout:-1}" piped_data; do
+        declare -a piped_versions
+        debug 10 "Versions piped to ${FUNCNAME}: ${piped_data}"
+        # shellcheck disable=2086
+        piped_versions+=( ${piped_data} )
+    done
+    shopt_decorator_option_name='nounset'
+    shopt_decorator_option_value='false'
+    # shellcheck disable=2015
+    shopt_decorator "${FUNCNAME[0]}" "${@:-}" && return || conditional_exit_on_fail 121 "Failed to run ${FUNCNAME[0]} with shopt_decorator"
+    # shellcheck disable=2068
+    _version_sort ${@} ${piped_versions[@]}
+}
+
+# Allows clear assert syntax
+function assert {
```

Comment:
Why move this?

Reply:
So that compare_versions can use assertions because it's called early on now that we use the shopt stuff.

Reply:
It was only broken the first time you imported the shtdlib :P